galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.ceda.manager.ivt/src/main/java/dev/galasa/cicsts/ceda/manager/ivt/CedaManagerIVT.java

```diff
@@ -203,49 +157,90 @@ public void testResourceLibrary() throws TextNotFoundException, ErrorTextFoundEx
+        assertThat(cics.ceda().resourceExists(cedaTerminal, resourceType, resourceName, groupName)).isEqualTo(true);
+
+        cics.ceda().installResource(cedaTerminal, resourceType, resourceName, groupName);
+        if (cics.cemt().inquireResource(cemtTerminal, resourceType, resourceName) != null) {
+           response = true;
+        }
+        assertThat(response).isEqualTo(true);
+			
+        // If resource was installed successfully, then tests the delete method by discarding resource from CEMT, 
+        // deleting and then trying to install and checking if the resource appeared on CEMT
+
+        if (response) {
+           response=false;
+           cics.cemt().discardResource(cemtTerminal, resourceType, resourceName);
+
+           if (cics.cemt().inquireResource(cemtTerminal, resourceType, resourceName) == null) {
+              cics.ceda().deleteResource(cedaTerminal, resourceType, resourceName, groupName);
+              cics.ceda().installResource(cedaTerminal, resourceType, resourceName, groupName);
+              if (cics.cemt().inquireResource(cemtTerminal, resourceType, resourceName) == null) {
+                 response = true;
+              }
+              assertThat(response).isEqualTo(true);
+
+              } else Fail.fail("Failed to discard resource");
+
+        } else Fail.fail("Failed to intsall / delete resource");
+
+     } catch (CedaException | CemtException e) {
+        e.printStackTrace();
+     }
+  }
+
+  @Test
+  public void testGroup() throws TextNotFoundException, ErrorTextFoundException, Zos3270Exception, InterruptedException, CicstsManagerException {
+     String resourceType = "prog";
+     String resourceName = "prg1";
+     String resourceName2 = "prg2";
+     String resourceName3 = "prg3";
+     String resourceName4 = "prg4";
+     String groupName = "IVT";
+     String groupName2 = "noIVT";
+     boolean result = false;
+     // Creating different resources in two different groups
+     cics.ceda().createResource(cedaTerminal, resourceType, resourceName, groupName, null);
+     cics.ceda().createResource(cedaTerminal, resourceType, resourceName2, groupName, null);
+     cics.ceda().createResource(cedaTerminal, resourceType, resourceName3, groupName, null);
+     /** Different group **/
+     cics.ceda().createResource(cedaTerminal, resourceType, resourceName4, groupName2, null);
+     // Installing only one group and check if installed group appeared in CEMT and not installed one did not
+     cics.ceda().installGroup(cedaTerminal, groupName);
+     if (cics.cemt().inquireResource(cemtTerminal,resourceType, resourceName).containsValue(resourceName.toUpperCase())&&cics.cemt().inquireResource(cemtTerminal,resourceType, resourceName2).containsValue(resourceName2.toUpperCase())&&cics.cemt().inquireResource(cemtTerminal,resourceType, resourceName3).containsValue(resourceName3.toUpperCase())&&cics.cemt().inquireResource(cemtTerminal,resourceType, resourceName4)==null) {

```

Comment:
can you re-factor this line to be a bit more readable please?

Reply:
I have refactored this to hopefully be more readable